plugins/sql-language/src/main/java/org/polypheny/db/sql/language/ddl/altermaterializedview/SqlAlterMaterializedViewAddIndex.java

```diff
@@ -118,25 +111,21 @@ public void unparse( SqlWriter writer, int leftPrec, int rightPrec ) {
 
 
     @Override
-    public void execute( Context context, Statement statement, QueryParameters parameters ) {
-        CatalogTable catalogTable = getCatalogTable( context, table );
-        if ( catalogTable.entityType != EntityType.MATERIALIZED_VIEW ) {
-            throw new RuntimeException( "Not Possible to use ALTER MATERIALIZED VIEW because " + catalogTable.name + " is not a Materialized View." );
-        }
-
+    public void execute( Context context, Statement statement, ParsedQueryContext parsedQueryContext ) {
+        LogicalTable catalogTable = getTableFailOnEmpty( context, table );
```

Comment:
Is there a reason for having a different method than searchEntity? However, in any case it needs a check whether the found entity is a materialized view and not a table.

Reply:
This is just a QoL wrapper around `searchEntity` to not pollute the SQL Node classes.

Added various error prevention and error messages for the users.